R/expr-lang.R

```diff
@@ -326,41 +356,43 @@ lang_fn <- function(call = caller_frame()) {
 #' lang_name(~foo$bar())
 #' lang_name(~foo[[bar]]())
 #' lang_name(~foo()())
-lang_name <- function(call = caller_frame()) {
-  call <- get_expr(call)
-  if (!is_lang(call)) {
-    abort("`call` must be a call or a tidy quote of a call")
+lang_name <- function(lang = caller_frame()) {
+  lang <- get_expr(lang)
+  if (!is_lang(lang)) {
+    abort("`lang` must be a call or must wrap a call (e.g. in a quosure)")
   }
 
-  switch_lang(call,
-    named = as_string(node_car(call)),
-    namespaced = as_string(node_cadr(node_cdar(call))),
+  switch_lang(lang,
+    named = as_string(node_car(lang)),
+    namespaced = as_string(node_cadr(node_cdar(lang))),
     NULL
   )
 }
 
-#' Return the head or tail of a call object.
+#' Return the head or tail of a call object
 #'
 #' @description
 #'
-#' Internally, calls are structured as a tree of expressions (see
-#' [switch_lang()] and [pairlist] documentation pages). A `language`
-#' object is the top level node of the tree. `lang_head()` and
-#' `lang_tail()` allow you to retrieve the node components.
+#' Internally, calls (`language` objects) are structured as a tree of
```

Comment:
This seems inconsistent with how we use "language" elsewhere. I think you need to somehow qualify this is what base R calls these things?